api/test/opentelemetry/trace/tracer_test.rb

```diff
@@ -0,0 +1,124 @@
+    end
+  end
+
+  let(:invalid_span) { Trace::Span::INVALID }
+  let(:tracer) { Tracer.new }
+
+  describe '#current_span' do
+    let(:current_span) { tracer.start_span('current') }
+
+    it 'returns an invalid span by default' do
+      tracer.current_span.must_equal(invalid_span)
+    end
+
+    it 'returns the current span' do
+      wrapper_span = tracer.start_span('wrapper')
+
+      tracer.with_span(wrapper_span) do
+        tracer.current_span.must_equal(wrapper_span)
+      end
+    end
+  end
+
+  describe '#in_span' do
+    it 'yields the new span' do
+      tracer.in_span('wrapper') do |span|
+        span.wont_equal(invalid_span)
+        tracer.current_span.must_equal(span)
+      end
+    end
+
+    it 'returns the result of the block' do
+      result = tracer.in_span('wrapper') { 'my-result' }
+      result.must_equal('my-result')
+    end
+
+    it 'finishes the new span at the end of the block' do
+      finish_tracer = TestInSpanFinishTracer.new
+      mock_span = nil
+      finish_tracer.in_span('wrapper') { |span| mock_span = span }
+      mock_span.verify
+    end
+  end
+
+  describe '#with_span' do
+    it 'yields the passed in span' do
+      wrapper_span = tracer.start_span('wrapper')
+
+      tracer.with_span(wrapper_span) do |span|
+        span.must_equal(wrapper_span)
+      end
+    end
+  end
+
+  describe '#start_root_span' do
+    it 'requires a name' do
+      proc { tracer.start_root_span(nil) }.must_raise(ArgumentError)
+    end
+
+    it 'returns a valid span' do
+      span = tracer.start_root_span('root')
+      span.context.must_be :valid?
+    end
+  end
+
+  describe '#start_span' do
+    let(:context) { Trace::SpanContext.new }
+    let(:invalid_context) { Trace::SpanContext::INVALID }
+    let(:parent) { tracer.start_span('parent') }
+
+    it 'requires a name' do
+      proc { tracer.start_span(nil) }.must_raise(ArgumentError)
+    end
+
+    it 'returns a valid span' do
+      span = tracer.start_span('op', with_parent_context: context)
+      span.context.must_be :valid?
+    end
+
+    it 'returns a span with a new context by default' do
+      span = tracer.start_span('op')
+      span.context.wont_equal(tracer.current_span.context)
+    end
+
+    it 'returns a span with the same context as the parent' do
+      span = tracer.start_span('op', with_parent: parent)
+      span.context.must_equal(parent.context)
+    end
+
+    it 'returns a span with a new context when passed an invalid context' do
+      span = tracer.start_span('op', with_parent_context: invalid_context)
+      span.context.wont_equal(invalid_context)
+    end
+  end
+
+  describe '#binary_format' do
+    it 'returns an instance of BinaryFormat' do
+      tracer.binary_format.must_be_instance_of(Propagation::BinaryFormat)
+    end
+  end
+
+  describe '#http_text_format' do
+    it 'returns an instance of HTTPTextFormat' do
+      tracer.http_text_format.must_be_instance_of(Propagation::HTTPTextFormat)
+    end
+  end
```

Comment:
These will move to `tracer_format_test.rb` once #122 lands.

Reply:
I've moved it.Thanks!